In [2]:
path = "C:\\Users\\mjjyo\\OneDrive\\Desktop\\REWE.txt"

In [3]:
addresses = []
# Getting my file w/ addresses
with open(path, 'r') as file:
    for line in file:
        if line.startswith("REWE Markt GmbH"):
            addresses.append(line.strip())


In [4]:
import re

# Preprocessing addresses
def is_valid_string(s):
    pattern = r'^[a-zA-Z0-9/.,!?"\'\s-]+$'
    return re.match(pattern, s) is not None

In [5]:
addresses = [address for address in addresses if is_valid_string(address) == True]

In [6]:
store = addresses[0]
store2 = addresses[1]

In [7]:
store2

'REWE Markt GmbH, Lausanner Str. 82 / Goerzallee, 12205 Berlin / Lichterfelde'

In [8]:
url = "https://www.rewe.de/angebote/berlin-friedenau/1350073/rewe-markt-rheinstr-30/"


https://www.rewe.de/angebote/berlin-friedenau/1350073/rewe-markt-rheinstr-30/
'REWE Markt GmbH, Lausanner Str. 82 / Goerzallee, 12205 Berlin / Lichterfelde'


In [9]:
from selenium import webdriver
from selenium.webdriver.common.keys import Keys
from selenium.webdriver.common.by import By
from selenium.webdriver.common.action_chains import ActionChains
# from selenium.webdriver.support import expected_conditions as EC
# from selenium.webdriver.support.ui import WebDriverWait
from selenium.common.exceptions import NoSuchElementException

import undetected_chromedriver as uc
import time

In [41]:
# Create a browser instance

options = uc.ChromeOptions()
options.headless = False

driver = uc.Chrome(options = options)

driver.get(url)

C:\Users\mjjyo\AppData\Local\Temp\ipykernel_7524\3501764175.py:4: DeprecationWarning: headless property is deprecated, instead use add_argument('--headless') or add_argument('--headless=new')
  options.headless = False


In [28]:
store_phones = []

In [43]:
# Click accept cookies
accept_cookies = driver.find_element(By.XPATH, "//button[@id='uc-btn-accept-banner']")
ActionChains(driver).move_to_element(accept_cookies).click().perform()

time.sleep(2.563)

# Change market button
market_prompt = driver.find_element(By.ID,'gbmc-js-cms-header-link')
ActionChains(driver).move_to_element(market_prompt).click().perform()

time.sleep(3.456)

# Select input and type in it
address_input = driver.find_element(By.XPATH, "//input[@data-testid='gbmc-text-input']")
ActionChains(driver).move_to_element(address_input).click().send_keys(store2).perform()

time.sleep(2.654)

# Click submit go to market button

go_to_market = driver.find_element(By.XPATH, "//button[@data-testid='gbmc-market-picker-4040098']")
ActionChains(driver).move_to_element(go_to_market).click().perform()

time.sleep(4.1265)

telephone_number = driver.find_element(By.XPATH, "//div[contains(@class, 'market-contact')]").text
store_phones.append(telephone_number)

In [51]:
store_phones[0].split('\n')[0].split(':')[1].strip()

'030-83202917'

In [ ]:
driver.close()